test/gen-server/lib/homedb/UsersManager.ts

```diff
@@ -0,0 +1,996 @@
+import { User } from 'app/gen-server/entity/User';
+import { AclRuleOrg } from 'app/gen-server/entity/AclRule';
+import { Group } from 'app/gen-server/entity/Group';
+import { Organization } from 'app/gen-server/entity/Organization';
+import { SUPPORT_EMAIL, UsersManager } from 'app/gen-server/lib/homedb/UsersManager';
+import { GetUserOptions, NonGuestGroup, Resource } from 'app/gen-server/lib/homedb/Interfaces';
+import { tmpdir } from 'os';
+import path from 'path';
+import { prepareDatabase } from 'test/server/lib/helpers/PrepareDatabase';
+import { prepareFilesystemDirectoryForTests } from 'test/server/lib/helpers/PrepareFilesystemDirectoryForTests';
+import { EnvironmentSnapshot } from 'test/server/testUtils';
+import { getDatabase } from 'test/testUtils';
+import { Workspace } from 'app/gen-server/entity/Workspace';
+import { Document } from 'app/gen-server/entity/Document';
+import { HomeDBManager } from 'app/gen-server/lib/homedb/HomeDBManager';
+import Sinon, { SinonSandbox, SinonSpy } from 'sinon';
+import { assert } from 'chai';
+import { FullUser, UserProfile } from 'app/common/LoginSessionAPI';
+import { ANONYMOUS_USER_EMAIL, EVERYONE_EMAIL, PREVIEWER_EMAIL, UserOptions } from 'app/common/UserAPI';
+import { Login } from 'app/gen-server/entity/Login';
+import { Pref } from 'app/gen-server/entity/Pref';
+import { EntityManager } from 'typeorm';
+import log from 'app/server/lib/log';
+import winston from 'winston';
+import { updateDb } from 'app/server/lib/dbUtils';
+
+const username = process.env.USER || "nobody";
+const tmpDir = path.join(tmpdir(), `grist_test_${username}_userendpoint`);
+
+describe('UsersManager', function () {
+  this.timeout(30000);
+
+  describe('static method', function () {
+    function* makeIdxIterator() {
```

Comment:
Can you move helper methods to the end of the file? (And even outside of the `describe` funcition)

Reply:
Ok, I don't have a strong opinion here. I just prefer to have them hidden, or not to have them at all :).